tests/sparseml/optim/test_helpers.py

```diff
@@ -15,13 +15,90 @@
 import pytest
 
 from sparseml.optim import (
+    check_if_staged_recipe,
     evaluate_recipe_yaml_str_equations,
     load_recipe_yaml_str,
     load_recipe_yaml_str_no_classes,
     update_recipe_variables,
 )
 
 
+STAGED_RECIPE = """
```

Comment:
looks like a good test case, let's get another one in with a few more edge cases like stage scope variable names conflicting with each other

Reply:
I have done so. The test fail to capture the logic which I would expected when variable names conflict. I would like to discuss this logic shortly.